src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,829 @@
+            SeparatedSyntaxList<SyntaxNode> invocationArguments, SyntaxGenerator generator,
+            SyntaxNode newArgumentExpression, int insertionIndex, string name, bool named)
+        {
+            var argument = named
+                ? generator.Argument(name, RefKind.None,
+                    newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation))
+                :
+                generator.Argument(newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
+
+            return invocationArguments.Insert(insertionIndex, argument);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a return expression of the highlighted expression.
+        /// Example:
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y; // highlight this expression
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public int M_f(int x, int y)
+        /// {
+        ///     return x * y;
+        /// }
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static async Task GenerateNewMethodAsync(Document document, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod, string newMethodIdentifier, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            // Remove trailing trivia because it adds spaces to the beginning of the following statement.
+            var newStatements = SpecializedCollections.SingletonEnumerable(editor.Generator.ReturnStatement(expression.WithoutTrailingTrivia()));
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetTypeInfo(expression, cancellationToken).ConvertedType ?? semanticModel.Compilation.ObjectType;
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, returnType: typeSymbol);
+
+            var newMethodDeclaration = editor.Generator.MethodDeclaration(newMethod, newStatements);
+            editor.InsertBefore(containingMethod, newMethodDeclaration);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter.
+        /// Example:
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y; // highlight this expression
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     M(x, y, x * y);
+        /// }
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y;
+        /// }
```

Comment:
this is an odd case.  you're generating an overload with the same parameters.  i don't really understand the example.

Reply:
n/m.  i see this is an intermediary step.  please doc that. 

first, please state which if these you are generating, and which already exists.  second, state that this is not hte final form.  but just a step along the way.

Reply:
added context